src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -572,6 +571,50 @@ public void execute() throws GitException, InterruptedException {
                     }
                 }
 
+                launchCloneCommand(urIish);
```

Comment:
This change will dramatically increase disc use and time to clone for large repositories which were using a reference repository previously.  It negates almost entirely the benefits of a reference repository by not adding the reference repository until after the clone has been performed.

The reference repository **must** be honored in the initial clone.  

If it is not honored, then large repositories (like the LInux kernel with its 2 GB repo or many other repos) will copy the entire repository on first clone.

Reply:
This change also removes all the benefits that were previously gained for Pipeline and multi-branch Pipeline jobs by honoring the refspec on first clone.  Honoring the refspec on first clone results in a significant reduction in clone time and disc use.

If a narrow refspec has been specified, it **must** be honored in the initial clone.

Reply:
This change also ignores the "no tags" setting on initial clone.  Honoring "no-tags" on initial clone has not been as dramatic a savings as honoring a narrow refspec or using a reference repository, but that would also be a regression for those jobs which have been created that do not clone tags initially.

Reply:
Are there cases where git init+fetch is faster than git clone? Is this reproducible using repositories like the linux kernel with specific refspecs?

Reply:
Yes, there are cases where init+fetch is much faster than clone.  My jenkins-bugs repo is only 50 MB (small in the world of large git repositories) and init+fetch is much faster than clone when using a narrow refspec.

Git init+fetch: less than 1 second for a single branch with a narrow refspec

```
$ mkdir jenkins-bugs
$ cd jenkins-bugs
$ git init
$ time git fetch git://github.com/MarkEWaite/jenkins-bugs.git +refs/heads/JENKINS-43198:refs/remotes/bare/JENKINS-43198
real    0m0.643s
user    0m0.016s
sys     0m0.016s
```

Git clone: 8 seconds

```
$ time git clone git://github.com/MarkEWaite/jenkins-bugs.git
real    0m8.748s
user    0m19.824s
sys     0m0.844s
```